src/components/mermaidInputs/InputWithLabelAndValidation/InputWithLabelAndValidation.js

```diff
@@ -102,10 +100,24 @@ const InputWithLabelAndValidation = ({
               checked={isCheckboxChecked}
               onChange={(event) => handleCheckboxChange(event.target.checked)}
             />
+
             {checkboxLabel}
           </CheckBoxLabel>
         ) : null}
-        {inputType}
+        <InputContainer>
```

Comment:
following up on my other comment. I wonder if here we just have a render slot and we can add any component to (and the logic for that component can live elsewhere. I cant see how this looks, but we could name that render slot based on its UI position. This adds theoretical flexibility, but also helps keep that logic defined elsewhere so that this generic component doesnt become too bloated.

eg `<InputWithLabelAndValidation leftItem={<AnyComponet doThis={x} isFoo={false}/>}`

Reply:
I need to think more on this one , as there are two things being rendered in different spots (checkbox and button) ... good to bring up though!

Reply:
cleaned this up - thanks for the feedback! removed checkbox and button from this component and just passed both as rendered items (within or above input)